tetris.py

```diff
@@ -220,44 +223,64 @@ def draw_block(self):
         }
         self.canvas.delete("block")
 
-        for x, y in self.get_block_shape():
-            self.draw_rectangle(x, y, "block", self.color_dict[self.block_letter])
-
-        for letter, coord_list in self.landed_blocks.items():
-            for (x, y) in coord_list:
-                self.draw_rectangle(x, y, "block", self.color_dict[letter])
+        if self.small_board is not None:
+            for letter, coord_list in landed_blocks.items():
+                for (x, y) in coord_list:
+                    self.draw_rectangle(x, y, "block", color_dict[letter])
+        else:
+            self.resize_to_fit(block_letter)
 
-    def new_block(self):
-        self.current_block_center = (int(game_width / 2), -2)
-        self.block_letter = self.small_board.block_letter
-        self.small_board.block_letter = random.choice(block_letters)
-        self.small_board.resize_to_fit()
-        self.small_board.draw_block()
-        self.rotate_counter = 0
-        self.fast_down = False
+        for x, y in block:
+            self.draw_rectangle(x, y, "block", color_dict[block_letter])
 
-    def resize_to_fit(self):
-        if self.block_letter == "L":
+    def resize_to_fit(self, block_letter):
+        if block_letter == "L":
             self.canvas.config(width=square_size * 3, height=square_size * 2)
-        if self.block_letter == "L_rev":
+        if block_letter == "L_rev":
             self.canvas.config(width=square_size * 3, height=square_size * 2)
-        if self.block_letter == "O":
+        if block_letter == "O":
             self.canvas.config(width=square_size * 2, height=square_size * 2)
-        if self.block_letter == "E":
+        if block_letter == "E":
             self.canvas.config(width=square_size * 3, height=square_size * 2)
-        if self.block_letter == "Z":
+        if block_letter == "Z":
             self.canvas.config(width=square_size * 3, height=square_size * 2)
-        if self.block_letter == "Z_rev":
+        if block_letter == "Z_rev":
             self.canvas.config(width=square_size * 3, height=square_size * 2)
-        if self.block_letter == "I":
+        if block_letter == "I":
             self.canvas.config(width=square_size * 4, height=square_size * 1)
-
-        if self.block_letter == "I":
             self.canvas.pack(pady=square_size / 2 + 10)
-            self.current_block_center = (2, 0)
         else:
             self.canvas.pack(pady=10)
-            self.current_block_center = (1, 1)
```

Comment:
I think the logic was more clear before. Are you sure you don't want a separate resizing part (everything that uses `self.canvas.config`) and padding part (everything that uses `self.canvas.pack`)?

Reply:
Maybe I should clarify what I meant.

Right now the code is like:

```
        ...
        if block_letter == "Z_rev":
            self.canvas.config(...)
        if block_letter == "I":
            self.canvas.config(...)
            self.canvas.pack(pady=square_size / 2 + 10)
        else:
            self.canvas.pack(pady=10)
```

Separating `config` and `pack` parts would make it easier to read:

```
        ...
        if block_letter == "Z_rev":
            self.canvas.config(...)
        if block_letter == "I":
            self.canvas.config(...)

        if block_letter == "I":
            self.canvas.pack(pady=square_size / 2 + 10)
        else:
            self.canvas.pack(pady=10)
```